Add org-inline-pdf recipe

shg opened the pull request:
### Brief summary of what the package does

Enables inline PDF previewing in Org buffers.

### Direct link to the package repository

https://github.com/shg/org-inline-pdf.el

### Your association with the package

I am the author and the maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks, this looks useful.  It may not be deliberate, but the only part of your `;;; Commentary` that will show up in <kbd>M-x finder-commentary</kbd> and MELPA  is the line "Enable inline PDF preview in Org buffers"; I think this is because it sees `;;; Usage` and thinks that's the end of the commentary.

If I run a clean byte-compile with Emacs 27.1:
```
org-inline-pdf.el:90:1:Warning: the function `org-babel-temp-file' might not
    be defined at runtime.
org-inline-pdf.el:90:1:Warning: the function `equalp' is not known to be
    defined.
```
`cl-equalp` is preferred to `equalp` (the latter is deprecated) and you'll have to `(require 'cl-lib)`.  The first warning is interesting -- this is the first time I can recall seeing it -- but you can silence the compiler by forward-declaring it with `(declare-function org-babel-temp-file "ob-core.el" ...)`

riscy commented:
Actually, now that I look, you may want to use `string=` instead of `equalp` and forget about `cl-equalp`.

shg commented:
@riscy Thank you for the review. I changed the commentary format, and addressed the compiler warnings.

The intention of using `equalp` was to do case-insensitive comparison. I rewrote it using `member` instead.

riscy commented:
Great, thanks.  Taking a closer look at the code, it looks like this package warrants a global minor mode that can be easily toggled off and on by users without restarting Emacs.  Generally we urge that simply loading a package should not modify Emacs' behavior (or otherwise do some arbitrary amount of work) except in a narrow set of cases (e.g. adding to auto-mode-alist).  Let me know if I can give you any pointers on that.

shg commented:
@riscy Thank you I understood your suggestion. I rewrote the code as a minor mode. I hope this conforms to the guidelines.

riscy commented:
Thank you!
